Leanplum/Classes/LPActionDefinition.m

```diff
@@ -32,4 +32,21 @@ - (id) initWithDictionary:(NSDictionary *)responseDict {
     return self;
 }
 
+- (id)initWithCoder:(NSCoder *)decoder {
```

Comment:
Why are we making this a protocol instead of `LPModel` base class with common implementation?

Reply:
I did not understand the question here. I had implemented this as LPModel base class with common implementation, and we did discuss that. But, later on i realized that it actually did not work as test cases were failing. As inheritance was top to bottom, the encoding and decoding was not storing values of the model. And these need to be implemented in the classes.

Reply:
There is definitely a way to make this work with the common implementation. The only thing that needs to be specific is the mapping and that can be a method that every superclass specifies.

>  You can check Mantle to see how its done

Reply:
yea i tried that and it failed.

Reply:
I tried this , and failed will discuss it with you . If we think there is an alternate way to refine this, lets definitely do that or do propose an idea or try it out. But lets get this PR merged to unblock, especially since you might be out.